### DISCUSSION TITLE
How to clip button texts?

### DISCUSSION THREAD

**Discussion by george-hopkins** [slint-ui/slint#4114] [Q&A]
I have a button which might contain a label which is wider than the window. Is it possible to cap the width of the button and truncate the label?
```
import { HorizontalBox, Button } from "std-widgets.slint";
export component Demo {
    HorizontalBox {
        width: 50px;

        Button {
            text: "A very long long text";
        }
    }
}
```
[on SlintPad](https://slint.dev/releases/1.3.2/editor/?snippet=import+%7B+HorizontalBox%2C+Button+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++HorizontalBox+%7B%0A++++++++width%3A+50px%3B%0A%0A++++++++Button+%7B%0A++++++++++++text%3A+%22A+very+long+long+text%22%3B%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)

I experimented with various width-related settings however I couldn't prevent the button from overflowing. My previous attempt to solve it in the core caused regressions: #3977

**Comment by bjorn** [comment]
I think the use-case for eliding long button texts is pretty rare and usually people don't want their buttons to resize. On the other hand, I think if you force the button to a small size, its label should probably elide. Possibly, your change just wasn't entirely complete, since it left the `min-width` of the button not set to a generally sensible default.

I'd suggest to resubmit your PR with a solution to this, for example, each `Button` could set its `min-width` as follows on the root item:

```css
min-width: self.preferred-width;
```

This way, the button should by default still refuse to shrink. However, in your app you can then override the `min-width` to whatever is still alright for your app, and when the button then needs to be smaller than its preferred size, the text would elide.

**Reply by ogoffart** [reply]
Yes, thanks @bjorn for your reply.
The button could still have a elide mode, if we make sure that it keeps the minimum width.
Then i think the minimum width can be overriden if we set an explicit width.
